data/card_definitions.json

```diff
@@ -6485,6 +6485,381 @@
 		"NOTEEND": "NAGO CARDS END"
 	},
 
+	{
+		"NOTE": "NOEL CARDS START",
+		"id": "noel_valkyrieveil",
+		"type": "ultra",
+		"display_name": "Valkyrie Veil",
+		"force_cost": 0,
+		"gauge_cost": 3,
+		"range_min": 1,
+		"range_max": 2,
+		"speed": 2,
+		"power": 3,
+		"armor": 5,
+		"guard": 0,
+		"effects": [
+			{
+				"timing": "during_strike",
+				"effect_type": "stun_immunity"
+			},
+			{
+				"timing": "before",
+				"condition": "was_hit",
+				"effect_type": "powerup",
+				"amount": 4,
+				"and": {
+					"effect_type": "choice",
+					"choice": [
+						{ "effect_type": "draw", "amount": 1 },
+						{ "effect_type": "draw", "amount": 2 },
+						{ "effect_type": "draw", "amount": 3 },
+						{ "effect_type": "draw", "amount": 4 }
+					]
+				}
+			}
+		],
+		"boost": {
+			"boost_type": "immediate",
+			"force_cost": 0,
+			"cancel_cost": -1,
+			"display_name": "Burst",
+			"effects": [
+				{
+					"timing": "immediate",
+					"effect_type": "draw",
+					"amount": 2
+				},
+				{
+					"timing": "immediate",
+					"effect_type": "seal_this"
+				},
+				{
+					"timing": "immediate",
+					"effect_type": "bonus_action"
+				}
+			]
+		}
+	},
+	
+	{
+		"id": "noel_bulletstorm",
+		"type": "ultra",
+		"display_name": "Bullet Storm",
+		"force_cost": 0,
+		"gauge_cost": 3,
+		"range_min": 3,
+		"range_max": 4,
+		"speed": 4,
+		"power": 6,
+		"armor": 0,
+		"guard": 0,
+		"effects": [
+			{
+				"timing": "during_strike",
+				"effect_type": "ignore_armor"
+			}
+		],
+		"boost": {
+			"boost_type": "immediate",
+			"force_cost": 0,
+			"cancel_cost": -1,
+			"display_name": "Power of the Eye",
+			"effects": [
+				{
+					"timing": "immediate",
+					"effect_type": "look_at_opponent_hand"
```

Comment:
I know this is kind of annoying, but can you use the "reveal_hand" effect we already have an add an extra field that is "opponent": true?
Then just have an if statement in the "reveal_hand" handling code that says:
if 'opponent' in effect and effect['opponent']:
    opposing_player.reveal_hand()
else:
    performing_player.reveal_hand()